testsuite/cluster-test/src/main.rs

```diff
@@ -616,6 +674,7 @@ impl ClusterTestRunner {
         mut experiment: Box<dyn Experiment>,
         mut global_emit_job_request: Option<EmitJobRequest>,
     ) -> Result<()> {
+        self.wait_until_all_healthy()?;
```

Comment:
Why is this needed here? Should it be part of deploy code?

Reply:
This is because we do k8s deployment during `ClusterUtil::setup` where AWS discovery happens and `ClusterUtil` does not have the `wait_until_all_healthy` function.

Generally speaking, once we migrate completely to k8s, we will probably have to rethink the testing models and workflows which cluster test will support and at that point I presume we will have a major refactor of components. Till then I want to avoid any major refactor to support both environments as a lot of the refactored code will be thrown away.